Add weekly, biweekly, and weekly/biweekly change options to Explorer

breck7 opened the pull request:
This change adds 8 more chart views + 8 more map views to the Explorer.

It also changes the Interval control to a dropdown to accommodate the new options.

It changes the dailyFreq=true, totalFreq=true query param booleans to a single enum type instead.

New:
![Screen Shot 2020-07-28 at 1 17 47 PM](https://user-images.githubusercontent.com/74692/88737570-bd8d0000-d0d4-11ea-874d-a96e56663d9f.png)

shaahmad commented on charts/covidDataExplorer/CovidExplorerTable.ts:
```diff
@@ -258,12 +260,7 @@ export class CovidExplorerTable {
-        rollingAverage?: number,
-        daily?: boolean
-    ): number {
+    private static buildCovidVariableId(params: CovidQueryParams): number {
```
Hmm, this makes it more difficult to override specific settings, and will make it a bit more difficult to build views (like the multimetric table) that differ from the URL.

breck7 replied:
Will change URL Params to ExplorerOptions in a later commit.

shaahmad commented on charts/covidDataExplorer/CovidChartUrl.ts:
```diff
@@ -322,16 +372,38 @@ export class CovidConstrainedQueryParams extends CovidQueryParams {
 
-    @computed get available() {
+    get rollingMultiplier() {
+        if (this.isWeekly) return 7
```
Would maybe be nice to isolate a "DataExplorerOptionsState" class from the URL class and have all these derivations in there and keep the URL class simple with just the state of the query params.

shaahmad commented on charts/covidDataExplorer/CovidChartUrl.ts:
```diff
@@ -58,20 +73,19 @@ export class CovidQueryParams {
         const combos: any = []
         metrics.forEach(metric => {
-            timelines.forEach(timeline => {
+            intervalOptions.forEach(interval => {
```
Nice to get rid of this "timeline" naming collision! Now we just need to resolve our two `Tables`... 😮

breck7 replied:
Where do you see two tables? I count at least 4 😮

garas commented:
Could it be option _Per million people_ available for _Weekly_ and _Biweekly_ options?

EU counties often uses _14-day cumulative number of COVID-19 cases per 100 000_ for evaluating the situation. _Biweekly per million people_ would be same value, just 10x, and for all world countries.

https://www.ecdc.europa.eu/en/cases-2019-ncov-eueea

breck7 commented:
Hi @garas,

Currently we disable certain combinations of options because either 1) the combination is hard to interpret (i.e. tests per case per million people) or 2) we haven't yet reviewed and adjusted all of the settings for a view (i.e. adjusting the map color scheme for a combination). For the weekly figures we haven't yet adjusted all of the settings for the map colors.

In the interim, I've added an option for advanced users to skip our constraints. Adding `everythingAvailable=true` to the query string will allow you to generate weekly/biweekly per capita charts (Example: https://ourworldindata.org/coronavirus-data-explorer?zoomToSelection=true&casesMetric=true&interval=weekly&perCapita=true&allowEverything=true&smoothing=0&country=USA~GBR~CAN~BRA~AUS~IND~DEU~MEX~ZAF~COL~KOR~NOR~NGA&pickerMetric=location&pickerSort=asc&everythingAvailable=true). Note that these charts aren't "supported" so the color schemes and so forth might not be ideal.

We don't yet support adjusting the per capita divisor to 100,000, but that would be something we could do at some point via a query string parameter.

Cheers,
Breck
